homeassistant/components/home_connect/select.py

```diff
@@ -243,7 +243,13 @@ async def async_update(self) -> None:
 
     async def async_select_option(self, option: str) -> None:
         """Select new program."""
-        bsh_key = PROGRAMS_MAP[option]
+        bsh_key = PROGRAMS_MAP.get(option)
+        if bsh_key is None:
```

Comment:
This can't happen if the options attribute are set only with items in `PROGRAMS_MAP`. We have validation in the base select entity that only options from the `options` attribute are valid.

Reply:
Options attribute is filled with the programs obtained from the endpoint but formatted to translation keys.
https://github.com/home-assistant/core/blob/ef257c289e8e97ddfa043cb8ee6dcead7d98ce2d/homeassistant/components/home_connect/select.py#L231-L233

Reply:
Right, but then it will fail before we get here, in the base class validation.

Reply:
Sorry, I see what you mean now.

We should discard the options that we don't have translation keys for already when setting the options attribute.